pandas/tests/test_strings.py

```diff
@@ -469,6 +469,62 @@ def test_replace_callable(self):
         exp = Series(['bAR', NA])
         tm.assert_series_equal(result, exp)
 
+    def test_replace_compiled_regex(self):
+        # GH 15446
+        values = Series(['fooBAD__barBAD', NA])
+
+        # test with compiled regex
+        pat = re.compile(r'BAD[_]*')
+        result = values.str.replace(pat, '')
+        exp = Series(['foobar', NA])
+        tm.assert_series_equal(result, exp)
```

Comment:
Thanks for tracing it out. I believe the issue is related to `re` in python 3.4 requiring an integer for `flags` (similar to the same requirement for 2.7). I'll look to fix it today.